### ISSUE TITLE
Cannot set fields to null with GraphQL

### ISSUE THREAD

**Issue by maxludovicohofer** [aws-amplify/amplify-category-api#75]

**Comment by IsaacTrevino** [comment]
I believe the VTL template should be written so `nullification` and `Delete` Operation are not the same. Delete should be able to delete an item from a table, and nullification should be part of an update, not the same thing.

This might sound drastic, but I would remove `nullAllowedFields` from VTL auth templates.

**Comment by CodySwannGT** [comment]
@IsaacTrevino agreed 100%. This is so bizarre. Having to re-write all my auth rules to include "delete", ie:

`{ allow: groups, groups: ["Admins"], operations: [read, update, create, delete] }`

Even though we explicitly remove the delete mutation via:

`@model(mutations: { update: "updateEvent", create: "createEvent" })`
